api/serializers.py

```diff
@@ -4,46 +4,63 @@
 from game.models import Competition, Team, Game, Score
 
 
-class CompetitionSerializer(serializers.HyperlinkedModelSerializer):
-    class Meta:
-        model = Competition
-        fields = ('id', 'name', 'description', 'slug')
-
-
-class TeamSerializer(serializers.HyperlinkedModelSerializer):
-    users = serializers.RelatedField(many=True)
-    name = serializers.Field(source='get_name')
-    competitions = serializers.Field(source='get_competitions')
+class ScoreSerializer(serializers.ModelSerializer):
+    team_name = serializers.ReadOnlyField(source='get_team_name')
+    team_avatar = serializers.ReadOnlyField(source='get_team_avatar')
+    team_id = serializers.ReadOnlyField()
 
     class Meta:
-        model = Team
-        fields = ('id', 'name', 'competitions', 'users')
-
+        model = Score
+        fields = ('id', 'team_id', 'team_name', 'team_avatar', 'score', 'stdev')
 
-class GameSerializer(serializers.Serializer):
-    winner_id = serializers.WritableField(required=False, write_only=True)
-    loser_id = serializers.WritableField(required=False, write_only=True)
-    competition_id = serializers.WritableField(required=False, write_only=True)
 
-    winner = serializers.WritableField(required=False, read_only=True)
-    loser = serializers.WritableField(required=False, read_only=True)
-    competitions = serializers.RelatedField(many=True, read_only=True)
+class CompetitionSerializer(serializers.ModelSerializer):
+    scores = ScoreSerializer(many=True, read_only=True)
 
     class Meta:
-        model = Game
+        model = Competition
+        fields = ('id', 'name', 'description', 'slug', 'scores')
 
 
-class ScoreSerializer(serializers.HyperlinkedModelSerializer):
-    team = serializers.RelatedField()
-    competition = serializers.RelatedField()
+class TeamSerializer(serializers.ModelSerializer):
+    name = serializers.ReadOnlyField(source='get_name')
 
     class Meta:
-        model = Score
-        fields = ('id', 'team', 'competition', 'score', 'stdev')
+        model = Team
+        fields = ('id', 'name', 'users', 'competitions')
+
 
+class UserSerializer(serializers.ModelSerializer):
+    def create(self, validated_data):
```

Comment:
In fact, there is no reason because we use viewsets.ReadOnlyModelViewSet in views. I'll remove the methods.